File: src/battleships-api/Models/Match.cs
namespace Models;

public class Match
{
    public string Name { get; set; }
    public bool IsPregame { get; set; }
    public List<Player> Players { get; set; }
    public MatchSettings Settings { get; set; }

    public Match(string name = "New match") {
        this.Name = name;
        this.IsPregame = true;
        this.Players = new();
        this.Settings = new MatchSettings();
    }
}

File: src/battleships-ui/src/models/Match.ts
import { Ammo } from './Ammo';
import MatchSettings from './MatchSettings';
import { Player } from './Player';

export class Match {
  name: string;
  isPregame: boolean;
  players: Player[];
  settings: MatchSettings = new MatchSettings();
  availableAmmoTypes: Ammo[] = [];

  constructor(object?: Partial<Match>) {
    this.name = object?.name ?? 'New match';
    this.isPregame = object?.isPregame ?? true;
    this.players = Player.mapList(object?.players);
    this.settings = new MatchSettings(object?.settings);
    this.availableAmmoTypes = Ammo.mapList(object?.availableAmmoTypes);
  }
}
